src/policies.rs

```diff
@@ -175,12 +176,46 @@ fn verify_commit_signatures<G: Git>(
     commits: &Vec<Commit<'_>>,
     fingerprints: &HashMap<String, Fingerprint>,
 ) -> Result<PolicyResult, Box<dyn Error>> {
+
+    //hashmap of VerificationCommits
+    let verification_commits: HashMap<_, _> =  commits.iter()
+        .map(|commit| {
+            let committer = commit.committer();
+            let committer_email = committer.email().map(|s| s.to_string()) ;
+            let fingerprint = committer_email.clone().and_then(|s| fingerprints.get(&s).map(|f| f.id.to_string()));
+            (commit.id().to_string(), 
+            VerificationCommit {
+                id : commit.id().to_string(),
+                email: committer_email,
+                is_identical_tree: G::is_identical_tree_to_any_parent(commit),
+                valid_signature: false,
+                fingerprint: fingerprint
+            } )
+    }).collect();
+
+    let repo_path = git.path();
+
+    //verify commit signatures in parallel
+    let checked_verification_commits:HashMap<_,_> =  verification_commits.par_iter().map(|(_k, v)|{
+            let valid_signature = G::verify_commit_signature(repo_path, v);
+            (v.id.to_string(), 
+            VerificationCommit {
```

Comment:
Check the example here: https://doc.rust-lang.org/book/ch05-01-defining-structs.html#creating-instances-from-other-instances-with-struct-update-syntax
If you're making a new one that's basically the old one but something changed, you can go
```
VerificationCommit {
  valid_signature: valid_signature.unwrap_or(false),
  ..v
}
```